Odata-docs/webapi-8/fundamentals/query-options.md

```diff
@@ -0,0 +1,562 @@
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.ApplyTo(Customers.AsQueryable());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+In the example above, the [ApplyTo](/dotnet/api/microsoft.aspnet.odata.query.odataqueryoptions.applyto) method applies to all query options.
+
+We call ApplyTo for each individual query as shown in the sub-sections below.
+
+### SelectExpand query
+We call `ApplyTo` on the `SelectExpand` property as follows:
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.SelectExpand.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+`SelectExpand` handles both `$select` and `$expand` query options.
+
+The `options.SelectExpand` property is an instance of `SelectExpandQueryOption`. It will be set when we make the request below:
+
+```http
+GET http://localhost:6285/odata/Customers?$expand=Orders
+```
+
+If we don't want to pass the query option in the request, but we want the `Orders` property to be expanded, we can initialize the `SelectExpandQueryOption` and set it in the request.
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    options.Request.ODataFeature().SelectExpandClause = new SelectExpandQueryOption(null, "Orders", options.Context,
+        new ODataQueryOptionParser(
+            options.Context.Model,
+            options.Context.NavigationSource.EntityType(),
+            options.Context.NavigationSource,
+            new Dictionary<string, string> { { "$expand", "Orders" } },
+            options.Context.RequestContainer)).SelectExpandClause;
+
+    IQueryable results = options.ApplyTo(Customers.AsQueryable());
+
+    return results as IQueryable<Customer>;
+}
+```
+
+When making the request, we don't need to use the `$expand` query option.
+
+```http
+GET http://localhost:6285/odata/Customers
+```
+
+Below will be the output:
+
+```json
+{
+    "@odata.context": "http://localhost:6285/odata/$metadata#Customers(Orders())",
+    "value": [
+        {
+            "Id": 1,
+            "Name": "Customer 1",
+            "Age": 31,
+            "Orders": [
+                {
+                    "Id": 1001,
+                    "Price": 10,
+                    "Quantity": 10
+                },
+                {
+                    "Id": 1002,
+                    "Price": 35,
+                    "Quantity": 2
+                }
+            ]
+        },
+        ...
+        ...
+    ]
+}
+```
+
+### Filter query
+We call `ApplyTo` on the `Filter` property as follows:
+
+```csharp
+public IQueryable<Customer> Get(ODataQueryOptions<Customer> options)
+{
+    IQueryable results = options.Filter.ApplyTo(Customers.AsQueryable(), new ODataQuerySettings());
```

Comment:
Refer to my other comment on overload of `ApplyTo` that doesn't expect an `ODataQuerySettings` argument

Reply:
The overload is not in `FilterQueryOption`.